docs/k8s.md

```diff
@@ -1,47 +1,151 @@
-The `example` folder contains a `k8s` example. Kubernetes is split into
+The [example folder](https://github.com/giantswarm/inago/tree/master/example) contains a simple [k8s example](https://github.com/giantswarm/inago/tree/master/example/k8s). The Kubernetes deployment is split into
 two groups: `k8s-master` and `k8s-node`.
 
 ## Note
 The deployment sets up a Kubernetes cluster, which uses insecure communication
 between the components. Use this only as a Demo or POC. If you want to deploy a
 Kubernetes cluster for production please modify the example, so secure
-communication is used. [Cluster TLS using OpenSSL](https://coreos.com/Kubernetes/docs/latest/openssl.html)
+communication is used: [Cluster TLS using OpenSSL](https://coreos.com/Kubernetes/docs/latest/openssl.html).
+Further, this deployment uses a single Kubernetes master, so the API is not highly available. For HA deployments of Kubernetes check out: [Building High-Availability Clusters](http://kubernetes.io/docs/admin/high-availability/).
 
 ## Deployment
-If your CoreOS cluster is up and running you can deploy Kubernetes in 4 steps:
 
-1. submit the k8s-master by running: `inagoctl submit k8s-master`
-2. submit the k8s-node by running: `inagoctl submit k8s-node`
-3. start the k8s-master by running: `inagoctl start k8s-master`
-4. start the k8s-node by running: `inagoctl start k8s-node`
+You can use this deployment on any CoreOS fleet cluster. Below we work with a 3 node cluster. For a simple way to setup a CoreOS cluster on your laptop try the official [Vagrant setup](https://coreos.com/os/docs/latest/booting-on-vagrant.html).
+Once your CoreOS cluster is up and running you can deploy Kubernetes in two easy steps:
 
-And you're done!
+First, we need to setup the network environment. This, needs to be done only once, but on all nodes, which is why we set this up to be a global unit (running on all machines).
 
-The k8s-node group's unit files contain the `Global=true` flag and will be scheduled on all machines.
+```
+$ inagoctl up k8s-network
+2016-04-14 15:31:05.273 | INFO     | context.Background: Succeeded to submit group 'k8s-master'.
+2016-04-14 15:32:01.532 | INFO     | context.Background: Succeeded to start group 'k8s-network'.
+```
+
+Then, we can schedule the k8s-master group.
+
+```
+$ inagoctl up k8s-master
+2016-04-14 15:33:09.396 | INFO     | context.Background: Succeeded to submit group 'k8s-master'.
+2016-04-14 15:34:01.558 | INFO     | context.Background: Succeeded to start group 'k8s-master'.
+```
+
+You can check if it is up with `inagoctl status k8s-master`.
+
+Now, we can schedule the nodes. As we have a 3 node cluster we'll be running 3 slices of nodes:
+
+```
+$ inagoctl up k8s-node 3
+2016-04-14 15:41:26.824 | INFO     | context.Background: Succeeded to submit group 'k8s-node'.
+2016-04-14 15:41:36.847 | INFO     | context.Background: Succeeded to start 3 slices for group 'k8s-node'
+```
+
+And we're done!
```

Comment:
The network one can be integrated with the master, but I thought it's cleaner outside as it's one-off and actually does not belong to k8s itself.

master and nodes need to be deployed separately as the master in this example is not HA (and not sliced).

Also, you would want some control over master separately from nodes as they are two different things. Even with an HA mster you would maybe want only 3 slices of the master, but you might want 10 nodes.